airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -38,7 +38,10 @@ def get_query(self, stream_slice: Mapping[str, Any]) -> str:
 
     def parse_response(self, response: SearchPager, stream_slice: Optional[Mapping[str, Any]] = None) -> Iterable[Mapping]:
         for result in response:
-            yield self.google_ads_client.parse_single_result(self.get_json_schema(), result)
+            try:
+                yield self.google_ads_client.parse_single_result(self.get_json_schema(), result)
+            except Unauthenticated as exception:
```

Comment:
Would it be better to add this error handling to `request_records_job` instead, as this error can occur during sending the request and not only parsing?

Reply:
added